analysis/analysis.py

```diff
@@ -0,0 +1,297 @@
+        gantt_rows.append(
+            {
+                "Task": task_name,
+                "Start": task_start_item[0],
+                "Finish": task_finish_item[0],
+                "Description": task_full_desc,
+            }
+        )
+        hops = ""
+        if task_name.startswith("Transfer"):
+            transfer_from = task_body_json["from"]
+            transfer_to = task_body_json["to"]
+            hops = str(abs(transfer_from - transfer_to))
+        table_rows.append(
+            {
+                "id": task_id,
+                "name": task_name,
+                "duration": calculate_duration(task_start_item[0], task_finish_item[0]),
+                "description": task_full_desc,
+                "hops": hops,
+            }
+        )
+        csv_rows.append(
+            [task_name, "", calculate_duration(task_start_item[0], task_finish_item[0]), hops]
+        )
+
+        # Only add subtasks for leafs of the tree
+        main_task_debug_string = "{0} {1}: {2}".format(str(task_bracket), task_name, task_desc)
+        if not has_more_specific_task_bracket(task_bracket, task_brackets):
+            subtasks = list(filter(lambda t: t[2]["id"] == task_id, stripped_content))
+            print("{0} - subtasks = {1}".format(main_task_debug_string, str(len(subtasks))))
+            print("----------------------------------------------------------------")
+            append_subtask(task_name, table_rows, csv_rows, subtasks)
+        else:
+            print(main_task_debug_string)
+            print("----------------------------------------------------------------")
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Raiden Scenario-Player Analysis")
+    parser.add_argument(
+        "input_file",
+        nargs="*",
+        help="File name of scenario-player log file as main input, if empty, stdin is used",
+    )
+    parser.add_argument(
+        "--output-csv-file",
+        type=str,
+        dest="csv_output_file",
+        default="raiden-scenario-player-analysis.csv",
+        help="File name of the CSV output file",
+    )
+    parser.add_argument(
+        "--output-gantt-file",
+        type=str,
+        dest="gantt_output_file",
+        default="raiden-scenario-player-analysis.html",
+        help="File name of the Gantt Chart output file",
+    )
+    parser.add_argument(
+        "--output-summary-file",
+        type=str,
+        dest="summary_output_file",
+        default="raiden-scenario-player-analysis.json",
+        help="File name of the summary output file",
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def main():
+    args = parse_args()
+
+    stripped_content = read_raw_content(args.input_file)
+
+    task_brackets = create_task_brackets(stripped_content)
+
+    gantt_rows = []
+    csv_rows = []
+    table_rows = []
+
+    fill_rows(gantt_rows, csv_rows, table_rows, task_brackets, stripped_content)
```

Comment:
Please don't use what some languages call "out" parameters.

Return things from the function and assign them to variables there.